### PR TITLE
Leadernolimit

### PR DESCRIPTION
**Pull request by ahmedqarmout2**

### CODE REVIEW

**Review by Mohamed-A-Mohamed — APPROVED** [review]

File: server/questionValidator.js [hunk 1]
```diff
@@ -20,12 +20,21 @@ const common = require('./common.js');
...
+ */
 var qTypeFailMsg = function(message) {
     return {success:false, err:message};
 }
 
-/*Validate all question fields on first entry to db*/
+/**
+ * Validate all question fields on first entry to db
+ *
+ * @param {*} info
```
Comment (esteeroramy) [inline server/questionValidator.js]: Add type instead of `*`

File: server/questionValidator.js [hunk 2]
```diff
@@ -39,18 +48,28 @@ exports.questionCreationValidation = function(info) {
     return validateQuestionAttributesByType(info,info.type);
 }
 
+/**
+ * validate default question values
+ *
+ * @param {*} questionData
```
Comment (esteeroramy) [inline server/questionValidator.js]: Type

File: server/questionValidator.js [hunk 3]
```diff
@@ -39,18 +48,28 @@ exports.questionCreationValidation = function(info) {
...
     }
     return successMsg;
 }
 
-/*Validate all fields that will be modified*/
+/**
+ * Validate all fields that will be modified
+ *
+ * @param {object} question
+ * @param {*} type
```
Comment (esteeroramy) [inline server/questionValidator.js]: Type

File: server/questionValidator.js [hunk 4]
```diff
@@ -79,6 +98,12 @@ exports.validateAttributeFields = function(question,type) {
     return successMsg;
 }
 
+/**
+ * validate question attributes by type
+ *
+ * @param {object} question
+ * @param {*} type
```
Comment (esteeroramy) [inline server/questionValidator.js]: Type

**Review by esteeroramy — CHANGES_REQUESTED** [review]
You are missing a lot of types and `@return` please add those

File: server/questionValidator.js (hunk 1 above)
Comment (ahmedqarmout2) [reply server/questionValidator.js]: There is no type for this, its any type that is passed in, there is no one type. Unless I am missing something

**Comment by esteeroramy** [comment]
Please add a picture of the settings

File: server/questionValidator.js [hunk 5]
```diff
@@ -200,7 +267,12 @@ var validateAllAttributesInGroup = function(objectToCheck, attributeType) {
     return true;
 }
 
-/*Validate an Array object to contain specific value types*/
+/**
+ * Validate an Array object to contain specific value types
+ *
+ * @param {*} arrayObject
```
Comment (esteeroramy) [inline server/questionValidator.js]: shouldnt this be array

File: server/questionValidator.js [hunk 6]
```diff
@@ -184,13 +239,25 @@ var orderingAttributeValidator = function(question) {
...
 var validateAttributeType = function(valueToCheck, key, attributeType) {
     return Object.prototype.toString.call(valueToCheck) === common.questionAttributes[attributeType][key].type;
 }
 exports.validateAttributeType = validateAttributeType;
 
-/*Validate all attributes in Object being passed and has correct field types*/
+/**
+ * Validate all attributes in Object being passed and has correct field types
+ *
+ * @param {*} objectToCheck
```
Comment (esteeroramy) [inline server/questionValidator.js]: object?

File: public/js/leaderboard.js [hunk 7]
```diff
@@ -75,13 +77,16 @@ var fetchLeaderboard = function() {
...
-    
-
     getCurrentStudent();
     $('#leaderboardBody').html('');
     $('#criteriaName').html(boardType.displayName);
     podiumScore = studentLeaderList[0][boardType.name];
     $('#topScore').html(podiumScore + ((boardType === leaderboardTypes.ACCURACYBOARD) ? '%' : ''));
+
+    if (!leaderboardLimited) {
+        leaderboardLimit = 99999;
```
Comment (ahmedqarmout2) [inline public/js/leaderboard.js]: I love how everyone is worried about the comments, but no one noticed that we have a fixed value here that makes no sense, and its bad coding. But yeah lets spend the day fixing the comments, that will help

Comment (esteeroramy) [reply public/js/leaderboard.js]: I mean you did it sooo......

**Review by esteeroramy — APPROVED** [review]

**Pull request merged**